integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -42,80 +42,137 @@ public static DatasetIdentifier fromCatalogTable(
       CatalogTable catalogTable, SparkSession sparkSession) {
     String tableName = nameFromTableIdentifier(catalogTable.identifier());
 
-    DatasetIdentifier di;
-    URI uri;
+    URI locationUri;
     if (catalogTable.storage() != null && catalogTable.storage().locationUri().isDefined()) {
-      uri = catalogTable.storage().locationUri().get();
-      di = fromURI(uri);
+      locationUri = catalogTable.storage().locationUri().get();
     } else {
-      uri = prepareUriFromDefaultTablePath(catalogTable, sparkSession);
-      di = fromURI(uri);
+      locationUri = getDefaultLocationUri(sparkSession, catalogTable.identifier());
     }
+    DatasetIdentifier locationDataset = fromURI(locationUri);
+    // perform normalization
+    locationUri = FilesystemDatasetUtils.toLocation(locationDataset);
+
+    Optional<DatasetIdentifier> symlinkDataset = Optional.empty();
 
     SparkContext sparkContext = sparkSession.sparkContext();
     SparkConf sparkConf = sparkContext.getConf();
     Configuration hadoopConf = sparkContext.hadoopConfiguration();
-    Optional<URI> metastoreUri = extractMetastoreUri(sparkContext);
+    Optional<URI> metastoreUri = getMetastoreUri(sparkContext);
     if (metastoreUri.isPresent()) {
       // dealing with Hive tables
       URI hiveUri = prepareHiveUri(metastoreUri.get());
-      DatasetIdentifier symlink = FilesystemDatasetUtils.fromLocationAndName(hiveUri, tableName);
-      return di.withSymlink(
-          symlink.getName(), symlink.getNamespace(), DatasetIdentifier.SymlinkType.TABLE);
+      symlinkDataset = Optional.of(FilesystemDatasetUtils.fromLocationAndName(hiveUri, tableName));
     } else if (catalogTable.provider().isDefined()
         && "hive".equals(catalogTable.provider().get())
         && "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory"
             .equals(
                 SparkConfUtils.findHadoopConfigKey(
                         hadoopConf, "hive.metastore.client.factory.class")
                     .orElse(""))) {
-      String region = System.getenv().get("AWS_DEFAULT_REGION");
-      if (region == null || region.isEmpty()) {
-        region = System.getenv("AWS_REGION");
+      symlinkDataset =
+          getGlueUri(sparkConf)
+              .map(uri -> FilesystemDatasetUtils.fromLocationAndName(uri, tableName));
+    } else {
+      Optional<URI> warehouseLocation =
+          getWarehouseLocation(sparkConf, hadoopConf)
```

Comment:
Actually, I'm not quite sure that in this case a warehouse location should be used. There probably should be location of Derby database, created on driver host by Spark.

Reply:
Without `enableHiveSupport` all table syntax is disabled, e.g. `CREATE TABLE` raises an exception.

Reply:
Hm, actually `CREATE TABLE (...)` and `CREATE TABLE (...) USING hive` are raising an exception, but `CREATE TABLE (...) USING parquet` does not.

Reply:
I can move this discussion to a separated issue, if required

Reply:
#2796